### PR TITLE
WINC-896, WINC-1162: Pick up openshift/kubernetes 1.29 rebase updates

### PR DESCRIPTION
**Pull request by jrvaldes**
Picks up openshift/kubernetes 1.29 rebase updates

### CODE REVIEW

**Comment by jrvaldes** [comment]
aws-e2e-operator job [passed](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/2000/pull-ci-openshift-windows-machine-config-operator-master-aws-e2e-operator/1743034005793542144)

**Comment by jrvaldes** [comment]
> /test aws-e2e-operator

@mansikulkarni96  this test passed before. See https://github.com/openshift/windows-machine-config-operator/pull/2000#issuecomment-1881013354

I'm awaiting for reviews (approval) to trigger the full test suite.

**Comment by jrvaldes** [comment]
/test remaining-required


to gather ci signal across platforms

File: pkg/nodeconfig/nodeconfig_test.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ func TestCreateKubeletConf(t *testing.T) {
...
..."C:\\\\k\\\\kubelet-ca.crt\"},\"webhook\":{\"cacheTTL\":\"0s\"},\"anonymous\":{\"enabled\":false}},\"authorization\":{\"webhook\":{\"cacheAuthorizedTTL\":\"0s\",\"cacheUnauthorizedTTL\":\"0s\"}},\"clusterDomain\":\"cluster.local\",\"clusterDNS\":[\"10.0.128.10\"],\"streamingConnectionIdleTimeout\":\"0s\",\"nodeStatusUpdateFrequency\":\"0s\",\"nodeStatusReportFrequency\":\"0s\",\"imageMinimumGCAge\":\"0s\",\"imageMaximumGCAge\":\"0s\",\"volumeStatsAggPeriod\":\"0s\",\"cgroupsPerQOS\":false,\"cpuManagerReconcilePeriod\":\"0s\",\"runtimeRequestTimeout\":\"10m0s\",\"maxPods\":250,\"kubeAPIQPS\":50,\"kubeAPIBurst\":100,\"serializeImagePulls\":false,\"evictionPressureTransitionPeriod\":\"0s\",\"featureGates\":{\"RotateKubeletServerCertificate\":true},\"memorySwap\":{},\"containerLogMaxSize\":\"50Mi\",\"systemReserved\":{\"cpu\":\"500m\",\"ephemeral-storage\":\"1Gi\",\"memory\":\"1Gi\"},\"logging\":{\"flushFrequency\":0,\"verbosity\":0,\"options\":{\"json\":{\"infoBufferSize\":\"0\"}}},\"enableSystemLogQuery\":true,\"shutdownGracePeriod\":\"0s\",\"shutdownGracePeriodCriticalPods\":\"0s\",\"containerRuntimeEndpoint\":\"npipe://./pipe/containerd-containerd\",\"enforceNodeAllocatable\":[]}",
```
Comment (sebsoto) [inline pkg/nodeconfig/nodeconfig_test.go:85]: This is only changing the test not the Kubelet configuration

Comment (sebsoto) [reply pkg/nodeconfig/nodeconfig_test.go:85]: Also the commit title says kubeconfig, when it should be kubelet config

**Review by sebsoto — CHANGES_REQUESTED** [review]
Thanks @jrvaldes mostly LGTM

File: pkg/nodeconfig/nodeconfig_test.go (hunk 1 above)
Comment (jrvaldes) [reply pkg/nodeconfig/nodeconfig_test.go:85]: Addressed.

Comment (sebsoto) [reply pkg/nodeconfig/nodeconfig_test.go:85]: This is still not doing what is described in the commit title
`Add ImageMaximumGCAge to kubelet conf `
The conf isnt being changed. Is that meant to happen or is this only a unit test change?

Comment (jrvaldes) [reply pkg/nodeconfig/nodeconfig_test.go:85]: this is only a unit test change, reworded the commit message

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

**Comment by mansikulkarni96** [comment]
@jrvaldes the azure-e2e-operator failure needs to be investigated.

**Comment by jrvaldes** [comment]
/hold

to investigate [issues](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_windows-machine-config-operator/2000/pull-ci-openshift-windows-machine-config-operator-master-azure-e2e-operator/1745259893402636288) with node creation in azure test

**Comment by jrvaldes** [comment]
/test remaining-required

to gather ci signal across cloud providers

**Comment by jrvaldes** [comment]
/test azure-e2e-operator 

ipi-install failed

**Comment by jrvaldes** [comment]
/test azure-e2e-operator

ipi-install failed

**Review by sebsoto — APPROVED** [review]

**Comment by jrvaldes** [comment]
/hold cancel

ci is looking greener with https://github.com/openshift/windows-machine-config-operator/pull/2000/commits/629fd94635a22cf044a049a0dc760b7819c245bc

**Pull request merged**